merino/utils/block_list.py

```diff
@@ -0,0 +1,3 @@
+"""Util module containing block list for Dynamic Wikipedia provider."""
+
+block_list: set[str] = set()
```

Comment:
Thoughts on using `frozenset`? This list will be immutable, so can restrict any changes by making it a `frozenset`. 

Also stylistic nit: this set will essentially be used like a constant, we should probably capitalize it to 
```suggestion
BLOCK_LIST: set[str] = set()
```